test/widgets/util/profile/profile_widget_test.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:motis_mitfahr_app/account/models/profile.dart';
+import 'package:motis_mitfahr_app/account/pages/profile_page.dart';
+import 'package:motis_mitfahr_app/account/widgets/avatar.dart';
+import 'package:motis_mitfahr_app/util/profiles/profile_widget.dart';
+
+import '../../../util/factories/profile_factory.dart';
+import '../../../util/mocks/mock_server.dart';
+import '../../../util/mocks/request_processor.dart';
+import '../../../util/mocks/request_processor.mocks.dart';
+import '../../../util/pump_material.dart';
+
+void main() {
+  final Profile profile = ProfileFactory().generateFake();
+  final MockRequestProcessor processor = MockRequestProcessor();
+
+  testWidgets('shows right information', (WidgetTester tester) async {
+    await pumpScaffold(tester, ProfileWidget(profile));
+
+    expect(find.text(profile.username), findsOneWidget);
+    final Finder avatar = find.byType(Avatar);
+    expect(avatar, findsOneWidget);
+    expect(tester.widget<Avatar>(avatar).profile, profile);
+
+    expect(find.text(profile.description!), findsNothing);
+
+    await pumpScaffold(tester, ProfileWidget(profile, showDescription: true));
+    expect(find.text(profile.description!), findsOneWidget);
+
+    await pumpScaffold(tester, ProfileWidget(profile, actionWidget: const Text('Action')));
+    expect(find.text('Action'), findsOneWidget);
+  });
+
+  testWidgets('navigates to ProfilePage', (WidgetTester tester) async {
+    MockServer.setProcessor(processor);
+    whenRequest(processor).thenReturnJson(profile.toJsonForApi());
+
+    await pumpScaffold(tester, ProfileWidget(profile));
+    await tester.tap(find.byType(ProfileWidget));
+    await tester.pumpAndSettle();
+
+    expect(find.byType(ProfilePage), findsOneWidget);
+  });
+
+  testWidgets('does not navigate to ProfilePage when not tappable', (WidgetTester tester) async {
+    await pumpScaffold(tester, ProfileWidget(profile, isTappable: false));
+    await tester.tap(find.byType(ProfileWidget), warnIfMissed: false);
```

Comment:
the `warnIfMissed` sounds weird... why is that needed? and how can the test work when the button is not successfully pressed?

Reply:
the point is that the button is not there since it's not tappable that's why i put warnIfMissed there

Reply:
siehe [hier](https://api.flutter.dev/flutter/flutter_test/WidgetController/tap.html)

Reply:
yes the only thing it does is surpress the warning i guess. I wasn't happy with that either put i did not find a better way to check that the widget is not tappable

Reply:
Okay yeah I guess that's fine. But maybe add a comment so this is immediately understandable

Reply:
and it's exactly what the flutter documentation recommends